### PR TITLE
doc: add customer p0 retrospective

### PR DESCRIPTION
**Pull request by dadlerj**
If you agree with these changes, will create issues for the todos here

cc @KattMingMing

### CODE REVIEW

File: doc/dev/retrospectives/customer_license_expiration.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Customer P0 retrospective: license expiration
```
Comment (nicksnyder) [inline doc/dev/retrospectives/customer_license_expiration.md]:
```suggestion
# Customer license expiration P0 retrospective 
```

File: doc/dev/retrospectives/customer_license_expiration.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+## Incident details
+
+* Date: Feb 21, 2019
+* Customer: https://app.hubspot.com/contacts/2762526/company/419771425
+* Discussion: https://sourcegraph.slack.com/archives/G9EN3TJDD/p1553176601007400
+
+## Why did this happen?
+
+Customer's contract originally provided them with an opt-out clause after 1 year. That 1 year mark fell on Mar 21, 2019.
```
Comment (nicksnyder) [inline doc/dev/retrospectives/customer_license_expiration.md]:
```suggestion
The customer's contract originally provided them with an opt-out clause after 1 year. That 1 year mark fell on Mar 21, 2019.
```

File: doc/dev/retrospectives/customer_license_expiration.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+## Action items
+
+What are we going to change to make sure customers get the correct license keys at the correct time?
+
+A few things. These changes all have the goal of making the process less manual/artisinal/custom.
+
+1) Dan will add code that creates a dismissable site alert for admins 7 days out and 1 day out from license expiration.
+
+1) Dan will add code (functional on Sourcegraph.com only) that sends an email to sales@sourcegraph.com whenever a license is 7 days away from expiration.
```
Comment (sqs) [inline doc/dev/retrospectives/customer_license_expiration.md]: A simpler thing for this point would be to add another page like https://sourcegraph.com/site-admin/dotcom/product/licenses that just shows licenses that are expiring soon (or have recently expired).

File: doc/dev/retrospectives/customer_license_expiration.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+After the payment was received, I didn't think to go to Sourcegraph.com to generate a new license key — I simply overlooked the second step. The deadline was still 1.5 months away, and the process I had gone through was purely financial/legal, not technical (e.g., I didn't recall that the original key was only for 1 year, rather than for the full 3 years).
+
+Fundamentally, the process is entirely manual today. It's up to the member of the GTM team who negotiates the contract to remember to send updated license keys out.
+
+## Action items
+
+What are we going to change to make sure customers get the correct license keys at the correct time?
+
+A few things. These changes all have the goal of making the process less manual/artisinal/custom.
```
Comment (sqs) [inline doc/dev/retrospectives/customer_license_expiration.md]: Also add an explicit step to audit all the current license keys to make sure we know which ones are in use, which are expiring soon, etc.

**Pull request merged**